integrationtests/helmops/controller/controller_test.go

```diff
@@ -0,0 +1,1006 @@
+			BeforeEach(func() {
+				targets = []fleet.BundleTarget{}
+				helmapp = getRandomHelmAppWithTargets("test-empty", targets)
+				// no chart is defined
+				helmapp.Spec.Helm.Chart = ""
+			})
+
+			It("does not create a bundle", func() {
+				Consistently(func(g Gomega) {
+					bundle := &fleet.Bundle{}
+					ns := types.NamespacedName{Name: helmapp.Name, Namespace: helmapp.Namespace}
+					err := k8sClient.Get(ctx, ns, bundle)
+					g.Expect(err).ToNot(BeNil())
+					g.Expect(errors.IsNotFound(err)).To(BeTrue(), err)
+				}, 5*time.Second, time.Second).Should(Succeed())
+			})
+		})
+
+		When("helmapp is added and then deleted", func() {
+			BeforeEach(func() {
+				targets = []fleet.BundleTarget{}
+				helmapp = getRandomHelmAppWithTargets("test-add-delete", targets)
+			})
+
+			It("creates and deletes the bundle", func() {
+				// bundle should be initially created
+				Eventually(func(g Gomega) {
+					bundle := &fleet.Bundle{}
+					ns := types.NamespacedName{Name: helmapp.Name, Namespace: helmapp.Namespace}
+					err := k8sClient.Get(ctx, ns, bundle)
+					g.Expect(err).ToNot(HaveOccurred())
+					t := []fleet.BundleTarget{
+						{
+							Name:         "default",
+							ClusterGroup: "default",
+						},
+					}
+					checkBundleIsAsExpected(g, *bundle, helmapp, t)
+				}).Should(Succeed())
+
+				// delete the helmapp resource
+				err := k8sClient.Delete(ctx, &helmapp)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				// eventually the bundle should be gone as well
+				Eventually(func(g Gomega) {
+					bundle := &fleet.Bundle{}
+					ns := types.NamespacedName{Name: helmapp.Name, Namespace: helmapp.Namespace}
+					err := k8sClient.Get(ctx, ns, bundle)
+					g.Expect(err).ToNot(BeNil())
+					g.Expect(errors.IsNotFound(err)).To(BeTrue(), err)
+				}).Should(Succeed())
+
+				// and the helmapp should be gone too (finalizer is deleted)
+				Eventually(func(g Gomega) {
+					fh := &fleet.HelmApp{}
+					ns := types.NamespacedName{Name: helmapp.Name, Namespace: helmapp.Namespace}
+					err := k8sClient.Get(ctx, ns, fh)
+					g.Expect(err).ToNot(BeNil())
+					g.Expect(errors.IsNotFound(err)).To(BeTrue(), err)
+				}).Should(Succeed())
+			})
+		})
+
+		When("version is not specified", func() {
+			BeforeEach(func() {
+				targets = []fleet.BundleTarget{}
+				helmapp = getRandomHelmAppWithTargets("test-no-version", targets)
+
+				// version is empty
+				helmapp.Spec.Helm.Version = ""
+				// reset secret, no auth is required
+				helmapp.Spec.HelmSecretName = ""
+
+				svr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					w.WriteHeader(http.StatusOK)
+					fmt.Fprint(w, helmRepoIndex)
+				}))
+				DeferCleanup(func() {
+					svr.Close()
+				})
+
+				// set the url to the httptest server
+				helmapp.Spec.Helm.Repo = svr.URL
+				helmapp.Spec.Helm.Chart = "alpine"
+			})
+
+			It("creates a bundle with the latest version it got from the index", func() {
```

Comment:
Nice one! Removing my comment on unit tests about the need to cover this use case :grin: 

Also: do we want a similar test case for a wildcard version?

Reply:
I've added the `*` case, not sure if in the best way, but I didn't want to repeat the test code.